packages/SwingSet/src/kernel/liveSlots.js

```diff
@@ -459,8 +465,8 @@ function build(
       } else {
         assert.fail(X`unrecognized slot type '${type}'`);
       }
-      registerValue(slot, val);
     }
+    registerValue(slot, val);
```

Comment:
We to make sure the `instanceKitMaker` function (the one userspace gives to `makeKind`) is invoked exactly once each time the virtual object is unserialized, regardless of whether there's already a Representative for it or not. Otherwise userspace can sense nondeterminism by counting the number of times their maker function is called.

I think this is at odds with having `convertSlotToVal` simply using the `slotToVal` table directly, since that will bypass `makeVirtualObjectRepresentative`. Maybe a workable approach would be:
* check for `virtual` before looking up `slotToVal`
* if it's virtual, call `makeVirtualObjectRepresentative()` to get a potential Representative
* *then* check `slotToVal`
  * if empty, promote the potentlal Representative to the real one, and register it with `registerValue`
  * if full, discard the potential Representative and use the registered one instead

We'll need a test, like the ones that use `gc()`, to show that the userspace function is invoked a deterministic number of times. Unserializing a parent virtual object (with our target VO in its `.state`), or receiving data from the kernel that references the VO, are two deterministic operations, and are thus allowed to influence something measurable from userspace, like the number of times their maker function is invoked. The state of a WeakRef is *not* a deterministic function of userspace behavior or kernel deliveries, so it cannot be allowed to influence anything measurable by userspace.